lib/spack/spack/directives.py

```diff
@@ -28,13 +28,23 @@ class OpenMpi(Package):
   * ``version``
   * ``requires``
   * ``redistribute``
+  * ``drop_all_conflicts``
```

Comment:
Once we are happy with the rest of this PR, we should look at extending it to `extends`, `provides`, `variant`, and maybe even `resource`. We should also definitely add a `drop_all_patches` directive, but that's trivial to add.